lib/https.js

```diff
@@ -182,8 +182,42 @@ exports.request = function(options, cb) {
   return http.request(options, cb);
 };
 
+exports.requestAsync = function(options) {
+  var resolveRequest = null;
+  const getResponse = new Promise((resolve) => {
+    resolveRequest = resolve;
+  });
+  const request = exports.request(options, (res) => {
```

Comment:
Why do we need to create extra callback wrapper here and not just pass `resolveRequest`?